great_expectations/data_context/store/gx_cloud_store_backend.py

```diff
@@ -542,6 +542,25 @@ def remove_key(self, key):
                 f"Unable to delete object in GX Cloud Store Backend: {repr(e)}"
             )
 
+    def _update(self, key, value, **kwargs):
+        existing = self._get(key)
+        if key[1] is None:
+            key = (key[0], existing["data"]["id"], key[2])
+
+        return self.set(key=key, value=value, **kwargs)
+
+    def _add_or_update(self, key, value, **kwargs):
+        try:
+            existing = self._get(key)
+        except StoreBackendError as e:
+            logger.info(f"Could not find object associated with key {key}: {e}")
+            existing = None
+        if existing is not None:
+            id = key[1] if key[1] is not None else existing["data"]["id"]
+            key = (key[0], id, key[2])
```

Comment:
I feel like this logic is pretty difficult to follow if you don't understand stores in depth (not a fault of this PR but more of an architectural thing) - hmm is there any way we can make this a bit clearer?

Reply:
Will add commentary in separate PR.